### PR TITLE
Kotlin migration of CardBrowserTest

### PR DESCRIPTION
**Pull request by raymondwzeng**
## Purpose / Description
Partially done to help test out #10779, also just to get more familiar with the tests in CardBrowser and the class itself.

## Approach
I pretty much followed the steps on the script, though I had to manually add the files as they weren't done so by git (not an issue on the end of the script, I just migrate files a bit differently).

## How Has This Been Tested?

Ran the tests individually, and made sure they outputted the same results.

CardBrowserKt

CardBrowserJava

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by puranjayK** [comment]
Hey, Kotlin conversion PR should have only 2 commits. 
Please squash the final commit so this PR only contains two commits:
- Rename
- Conversion

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+        val b = browserWithNoNewCards
+        b.filterByTag("sketchy::(1)")
+        advanceRobolectricLooperWithSleep()
+        MatcherAssert.assertThat("tagged card should be returned", b.cardCount, Is.`is`(1))
+    }
+
+    @Test
+    fun filterByFlagDisplaysProperly() {
+        val cardWithRedFlag = addNoteUsingBasicModel("Card with red flag", "Reverse")
+        flagCardForNote(cardWithRedFlag, 1)
```
Comment (Prince-kushwaha) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:337]: Could you fix the spacing of the code to match how it previously looked in `CardBrowserTest.java`

File: AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+        advanceRobolectricLooperWithSleep()
+        MatcherAssert.assertThat("Flagged cards should be returned", b.cardCount, Is.`is`(2))
+    }
+
+    @Test
+    fun previewWorksAfterSort() {
+        // #7286
+        val cid1 = addNoteUsingBasicModel("Hello", "World").cards()[0].id
+        val cid2 = addNoteUsingBasicModel("Hello2", "World2").cards()[0].id
+        val b = browserWithNoNewCards
```
Comment (Prince-kushwaha) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardBrowserTest.kt:353]: Could you fix the spacing of the code to match how it previously looked in CardBrowserTest.java

**Review by Prince-kushwaha — CHANGES_REQUESTED** [review]

**Comment by Prince-kushwaha** [comment]
@raymondwzeng follow this for kotlin migration https://github.com/ankidroid/Anki-Android/blob/main/docs/kotlin-migration.md

**Comment by Arthur-Milchior** [comment]
Thanks @Prince-kushwaha for the review. However, in this case, since this test migration script on windows, this link is not relevant here.

@raymondwzeng Do you by any luck still have the console output, so that I can know what failed? Or otherwise, can you run the script again.
In particular, there is indeed a problem. In the first commit, the java file should have been renamed to .kt  (with no content changed), and it's not normal that you don't have renaming in the first commit (plus, this first commit don't compile, which is not acceptable for us to merge it in our branch)

**Comment by raymondwzeng** [comment]
Going to redo this PR later today + include the comments from here. Seems like something went wrong along the way when I decided to squash some merge commits.

Also per Arthur's comment, something may have gone wrong with my process in terms of migrating using the script. I'll tack that and additional error info on #10779.

**Pull request closed without merging**